internal/configconverter/normalize_gcp.go

```diff
@@ -0,0 +1,78 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configconverter
+
+import (
+	"context"
+	"fmt"
+	"go.opentelemetry.io/collector/confmap"
+	"log"
+)
+
+type NormalizeGcp struct{}
+
+func (NormalizeGcp) Convert(_ context.Context, in *confmap.Conf) error {
+	if in == nil {
+		return fmt.Errorf("cannot NormalizeGcp on nil *confmap.Conf")
+	}
+
+	const resourceDetector = "processors::resourcedetection::detectors"
+	out := map[string]any{}
+	nonNormalizedGcpDetectorFound := false
+	for _, k := range in.AllKeys() {
+		v := in.Get(k)
+		match := resourceDetector == k
+		if !match {
+			out[k] = v
+		} else {
+			switch v.(type) {
+			case string:
```

Comment:
detectors have always been a yaml list so would this ever be a string? https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/dfcd49a8cbf68f10a4d77aff453c2e1e6a5256b3#diff-6c201bcf6478be7ec71d2ed94132853dfa188e033c9c432048d19744ffdf2553R28